edit-post/components/header/plugin-sidebar-more-menu-item/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * WordPress dependencies
+ */
+import { compose, createHigherOrderComponent } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { MenuItem } from '@wordpress/components';
+import { PluginContext } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import PluginsMoreMenuGroup from '../plugins-more-menu-group';
+
+const PluginSidebarMoreMenuItem = ( { children, isSelected, icon, onClick } ) => (
+	<PluginsMoreMenuGroup>
+		{ ( fillProps ) => (
+			<MenuItem
+				icon={ isSelected ? 'yes' : icon }
+				isSelected={ isSelected }
+				onClick={ compose( onClick, fillProps.onClose ) }
+			>
+				{ children }
+			</MenuItem>
+		) }
+	</PluginsMoreMenuGroup>
+);
+
+export default compose(
+	createHigherOrderComponent(
+		( OriginalComponent ) => ( props ) => (
+			<PluginContext.Consumer>
+				{ ( { pluginName } ) => (
+					<OriginalComponent
+						{ ...props }
+						pluginName={ pluginName }
+					/>
+				) }
+			</PluginContext.Consumer>
+		),
+		'withPluginContext'
+	),
```

Comment:
Is there a plan to reuse this? (In theory, `withPluginContext` does sound like something we'd expect `@wordpress/plugins` to offer.) Otherwise, I worry it may hurt readability of the module when compared with

```js
const PluginsSidebarMoreMenuItem = ( { children, isSelected, icon, onClick } ) => (
  <PluginContext.Consumer>
    { ( { pluginName } ) => (
      <PluginsMoreMenuGroup>
        …
```